spec/requests/comments_spec.rb

```diff
@@ -0,0 +1,314 @@
+# frozen_string_literal: true
+
+describe 'Comments', type: :request do
+  describe '#create' do
+    context 'when a user is not signed in' do
+      before { post comments_path }
+      it_behaves_like :with_no_logged_in_user
+    end
+
+    context 'when a user is signed in' do
+      subject { post comments_path, params: { comment: comment_params } }
+
+      before { sign_in user }
+
+      let(:user) { create(:user) }
+
+      # Shared examples reduce repeated code
+      shared_examples_for 'a comment is not created' do
+
+        it { is_expected.to eq 400 }
+
+        it 'returns an empty comment' do
+          subject
+          expect(JSON.parse(response.body)).to eq({})
```

Comment:
```suggestion
          expect(JSON.parse(response.body)).to be_empty
```

Just a nitpick otherwise this is fine too.

Reply:
Ahhh good point 👍 if I get back to my PC before it's merged I'll fix it, it not maybe if I can do another rspec refactor i'll do it then... Or maybe we can do a like, once over refactor once all of the smaller issues are merged 🤔